packages/smooth_app/lib/pages/product/attribute_icons.dart

```diff
@@ -0,0 +1,622 @@
+        ),
+      'low_salt' => _AttributeSaltIcon(
+          backgroundColor: backgroundColor,
+          foregroundColor: foregroundColor,
+          size: size,
+          semanticsLabel: semanticsLabel,
+        ),
+      'low_saturated_fat' => _AttributeSaturatedFatIcon(
+          backgroundColor: backgroundColor,
+          foregroundColor: foregroundColor,
+          size: size,
+          semanticsLabel: semanticsLabel,
+        ),
+      'low_sugars' => _AttributeSugarIcon(
+          backgroundColor: backgroundColor,
+          foregroundColor: foregroundColor,
+          size: size,
+          semanticsLabel: semanticsLabel,
+        ),
+      'nova' => _AttributeNOVAIcon(
+          backgroundColor: backgroundColor,
+          foregroundColor: foregroundColor,
+          size: size,
+          semanticsLabel: semanticsLabel,
+        ),
+      'palm_oil_free' => _AttributePalmOilIcon(
+          backgroundColor: backgroundColor,
+          foregroundColor: foregroundColor,
+          size: size,
+          semanticsLabel: semanticsLabel,
+        ),
+      'vegan' => _AttributeVeganIcon(
+          backgroundColor: backgroundColor,
+          foregroundColor: foregroundColor,
+          size: size,
+          semanticsLabel: semanticsLabel,
+        ),
+      'vegetarian' => _AttributeVegetarianIcon(
+          backgroundColor: backgroundColor,
+          foregroundColor: foregroundColor,
+          size: size,
+          semanticsLabel: semanticsLabel,
+        ),
+
+      /*'nutriscore' => EMPTY_WIDGET,
+      'ecoscore' => EMPTY_WIDGET,*/
+      _ => throw UnsupportedError('Unsupported attribute: ${attribute.id}'),
+    };
+  }
+
+  const AttributeIcon._({
+    required this.icon,
+    required this.backgroundColor,
+    required this.size,
+    this.iconSizeFactor,
+    this.angle,
+    this.offsetFactor,
+    Color? foregroundColor,
+    this.paddingFactor,
+    this.semanticsLabel,
+    this.clip = false,
+  }) : foregroundColor = foregroundColor ?? Colors.white;
+
+  final Widget icon;
+  final Color backgroundColor;
+  final Color foregroundColor;
+  final double size;
+  final double? angle;
+  final Offset? offsetFactor;
+  final double? iconSizeFactor;
+  final EdgeInsetsDirectional? paddingFactor;
+  final String? semanticsLabel;
+  final bool? clip;
+
+  @override
+  Widget build(BuildContext context) {
+    Widget child = Semantics(
+      label: semanticsLabel,
+      excludeSemantics: true,
+      child: DecoratedBox(
+        decoration: ShapeDecoration(
+          shape: const CircleBorder(),
+          color: backgroundColor,
+        ),
+        child: Padding(
+          padding: paddingFactor != null
+              ? EdgeInsetsDirectional.only(
+                  start: paddingFactor!.start * size,
+                  end: paddingFactor!.end * size,
+                  top: paddingFactor!.top * size,
+                  bottom: paddingFactor!.bottom * size,
+                )
```

Comment:
Just checking: does it mean that the display will be the same for LTR and RTL?

Reply:
Mmm I don't really understand your point.
The idea of **Directional objects is to manage LTR and RTL automatically, by flipping left & right

Reply:
My point here is that for some reason, you sometimes decide to move the icon slightly to the left or to the right, so that it looks better.
That would mean that you say `start` instead of `left`, the icon would be moved in the opposite direction, and would look worse.